### ISSUE TITLE
Click on element not works when component has changeDetectionRef.detach() call

### ISSUE THREAD

**Issue by franklbt** [angular/angular#43623]
### Which @angular/* package(s) are the source of the bug?

core

### Is this a regression?

No

### Description

For the needs of my project, I created a custom table component. This component has the particularity of functioning as a template. That is to say that each content of a cell is an ngTemplateOutlet which refers to a template. This component is also composed of two subcomponents: one for rows (table-row.component) and another for cells (table-cell.component). Knowing that the table component contains a list of rows, each row contains a list of cells, and each cell has a reference to a template. Another particularity is that for performance reasons, I added a call to ChangeDetectorRef.detach () in the ngOnInit of these 3 components, in order to finely control which cell must refresh its UI at the right time.

Here is a short version of the code describing the implementation:

**table.component.html**

```
<div  
  #tableContainer
  cdkScrollable
>
  <ng-container>
    <cdk-virtual-scroll-viewport>
      <table>
        <ng-container [ngTemplateOutlet]="tmplTableHeads"></ng-container>
        <tbody>
          <wr-table-row style="display: contents"
                        *cdkVirtualFor="let rowItem of paginatedRows; let i = index"
                        [data]="rowItem" [index]="i" [table]="this"></wr-table-row>
        </tbody>
      </table>
    </cdk-virtual-scroll-viewport>
  </ng-container>
</div>
```

**table.component.ts**

```
@Component({
  selector: 'wr-table',
  templateUrl: './table.component.html',
  styleUrls: ['./table.component.scss']
})
export class TableComponent implements OnInit {
  @Input() heads: TableHead[];
  @Input() rows: TableRow[];
  @ViewChildren(TableRowComponent) public rowsComponents: QueryList<TableRowComponent>;

  constructor(
    public cdRef: ChangeDetectorRef
  ) {
  }

  ngOnInit(): void {
    this.cdRef.detach();
  }

  detectChangeOnCell(rowIndex, columnIndex) {
    this.rowsComponents.get(rowIndex).cellsComponents.get(columnIndex).detectChanges();
  }
}
```

**table-row.component.ts**

```
<tr> 
  <ng-container *ngFor="let head of table.heads; let i = index">
    <wr-table-cell style="display: contents" [head]="head" [columnIndex]="i" [rowIndex]="index" [data]="data" [table]="table"></wr-table-cell>
  </ng-container>
</tr>
```

**table-row.component.ts**

```
@Component({
  selector: 'wr-table-row',
  templateUrl: './table-row.component.html'
})
export class TableRowComponent implements AfterViewChecked {
  @Input() table: TableComponent;
  @Input() data: any;
  @Input() index: number;

  @ViewChildren(TableCellComponent) cellsComponents: QueryList<TableCellComponent>;

  constructor(private cdRef: ChangeDetectorRef) {
  }

  ngAfterViewChecked(): void {
    this.cdRef.detach();
  }

  detectChanges() {
    this.cdRef.detectChanges();
  }
}
```

**table-cell.component.html**

```
<td>
  <div>
    <ng-container
      [ngTemplateOutlet]="table.headsTmpl[columnIndex]" 
      [ngTemplateOutletContext]="{
         data: data[head.id],
         row: data,
         justify: head.justifyContent,
         headId: head.id,
         columnIndex: columnIndex,
         rowIndex: rowIndex
      }"
    >
    </ng-container>
  </div>
</td>
```

**table-cell.component.ts**

```
@Component({
  selector: 'wr-table-cell',
  templateUrl: './table-cell.component.html'
})
export class TableCellComponent implements OnInit, AfterViewChecked {
  @Input() table: TableComponent;
  @Input() data: any;
  @Input() columnIndex: number;
  @Input() head: TableHead;
  @Input() rowIndex: number;

  constructor(private cdRef: ChangeDetectorRef) {
  }

  ngOnInit(): void {
  }

  ngAfterViewChecked(): void {
    this.cdRef.detach();
  }

  detectChanges(){
    this.cdRef.detectChanges();
  }
}
```

The problem is that in one of my templates attached to a cell I call `(click)="detectChangeOnCell(rowIndex, columnIndex)"`, and I put a breakpoint in this method, it is never reached . This means that in this configuration the evevement `(click)` does not work.

### Please provide the environment you discovered this bug in

```true
Angular CLI: 11.2.13
Node: 14.16.0
OS: win32 x64
NodeJS: v14.16.0
npm: v14.16.0

Angular: 11.2.14
... animations, common, compiler, compiler-cli, core, forms
... language-service, localize, platform-browser
... platform-browser-dynamic, router
Ivy Workspace: Yes

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1102.13
@angular-devkit/build-angular   0.1102.13
@angular-devkit/core            11.2.13
@angular-devkit/schematics      11.2.13
@angular/cdk                    11.2.12
@angular/cli                    11.2.13
@angular/material               11.2.12
@schematics/angular             11.2.13
@schematics/update              0.1102.13
rxjs                            6.6.7
typescript                      4.0.7
```

**Comment by pkozlowski-opensource** [comment]
Could you please provide a smaller and runable (in stackblitz or similar) reproduce scenario? It is very hard to follow the shared code without ability to run. Also please try to simplify it as much as possible.
